service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageHearingsService.java

```diff
@@ -282,6 +293,46 @@ public Object getSelectedDynamicListType(CaseData caseData) {
         return caseData.getPastAndTodayHearingDateList();
     }
 
+    public Map<String, Object> populateFieldsWhenPastDateAdded(CaseData caseData) {
+        Map<String, Object> data = new HashMap<>();
+
+        if (caseData.getHearingEndDate().isBefore(LocalDateTime.now()) && caseData.getHearingStartDate()
```

Comment:
Extract caseData.getHearingEndDate and caseData.getHearingStartDate to variables. 

Might be worth considering smaller methods on caseData

Reply:
will keep as variables for now